routes/signUpRoute.js

```diff
@@ -0,0 +1,147 @@
+    }
+    else{
+        //hash with salting add random string 
+        bcrypt.hash(req.body.password, 10, async(err,hash) =>{
+        if(err){
+            return res.status(statusCode.forbidden.code).json({
+                error:errorMsg.forbidden.message
+            });
+        }
+        else{
+            const token = jwt.sign({name : req.body.name, email: req.body.email}, secretKey.token.key, { expiresIn: '24h' })
+            const refreshToken = await jwt.sign({name : req.body.name, email: req.body.email},secretKey.token.key,{expiresIn: '30d'})
+            try{
+                const user = new User({
+                    name: req.body.name,
+                    email: req.body.email,
+                    password: hash,
+                    refreshToken : refreshToken
+                })
+                user.save()
+                const response = {
+                    "status": "Logged in",
+                    "token": token,
+                    "refreshToken": refreshToken,
+                }
+                tokenList[refreshToken] = response
+                res.status(statusCode.ok.code).json({data : user, tokenResponse: response})
+            }
+            catch(err){
+                next(err)
+                res.json(err)
+            }
+        } 
+    })
+    }    
+});
+
+router.post('/users', verifyToken, (req,res)=>{
+    
+    const bearerHeader = req.headers['authorization'];
+    jwt.verify(req.token,secretKey.token.key, (err,authData)=>{
+        if(err) res.status(statusCode.forbidden.code).json(errorMsg.forbidden.message)
+        else{   
+            res.json({
+                message: 'OK good',
+                authData: authData
+            })
+        }
+    })
+})
+
+
+router.post('/login',async(req,res) => {
+    let user =  await User.findOne({email: req.body.email})
+    if(!user) res.json({message:errorMsg.auth_fail})
+    var result = await bcrypt.compare(req.body.password, user.password)
+    if(!result) {
+        return res.status(statusCode.forbidden.code).json({message:errorMsg.forbidden.message})
+    }
+    if(result) {
+        const token = jwt.sign({name : req.body.name, email: req.body.email}, secretKey.token.key, { expiresIn: '24h' })
+        const refreshToken = jwt.sign({name : req.body.name, email: req.body.email},secretKey.token.key,{expiresIn: '30d'})
+        const response = {
+            "status": "Logged in",
+            "token": token,
+            "refreshToken": refreshToken,
+        }
+        await User.updateOne({'email' : req.body.email },
+        { $push: { 'refreshToken' : refreshToken } })
+        tokenList[refreshToken] = response
+        res.status(statusCode.ok.code).json(response);
+    }
+})
+
+
+router.post('/logout', verifyToken, async(req,res)=>{
+    let user = await User.findOne({refreshToken:req.header})
+    jwt.verify(req.token,secretKey.token.key, async(err,authData)=>{
```

Comment:
@AwaleRohin add functions to verify the authorization token in middleware. If you do that you don't have to verify everywhere